Packages/com.thenathannator.hidrogen/HIDrogen/Backends/USB/X360Controller.cs

```diff
@@ -0,0 +1,340 @@
+                    subType = subType,
+                    flags = XInputDeviceFlags.Wireless | XInputDeviceFlags.Voice,
+                    gamepad = new XInputGamepad(),
+                    vibration = new XInputVibration()
+                }),
+            };
+
+            m_Backend.QueueDeviceAdd(description, new USBQueueContext()
+            {
+                userIndex = _userIndex
+            });
+
+            switch (_userIndex)
+            {
+                case 0: {
+                    SetLED(LEDState.LED_1);
+                    break;
+                }
+                case 1: {
+                    SetLED(LEDState.LED_2);
+                    break;
+                }
+                case 2: {
+                    SetLED(LEDState.LED_3);
+                    break;
+                }
+                case 3: {
+                    SetLED(LEDState.LED_4);
+                    break;
+                }
+            }
+        }
+
+        public void SendInquiry()
+        {
+            var payload = new byte[12];
+            payload[0] = 8;
+            payload[2] = 0x0f;
+            payload[3] = 0xc0;
+
+            _receiver.InterruptTranferOut(controlEndpoint, payload);
+        }
+
+        public void PowerDown()
+        {
+            var payload = new byte[12];
+            payload[2] = 8;
+            payload[3] = 0xc0;
+            _receiver.InterruptTranferOut(controlEndpoint, payload);
+        }
+
+        public void SetLED(LEDState ledState)
+        {
+            var payload = new byte[12];
+            payload[2] = 8;
+            payload[3] = (byte)((uint)ledState + 0x40);
+            _receiver.InterruptTranferOut(controlEndpoint, payload);
+        }
+
+        public void SetRumble(byte highFreq, byte lowFreq)
+        {
+            var payload = new byte[12];
+            payload[1] = 1;
+            payload[2] = 0x0f;
+            payload[3] = 0xc0;
+            payload[5] = highFreq;
+            payload[6] = lowFreq;
+
+            _receiver.InterruptTranferOut(controlEndpoint, payload);
+        }
+
+        public void Dispose()
+        {
+            if (connected)
+            {
+                PowerDown();
+                m_Backend.QueueDeviceRemove(device);
+            }
+
+            _receiver.ReleaseInterface(interfaceIndex);
+            connected = false;
+        }
+
+        public void HIDUpdate(byte[] bytes)
+        {
+            XInputGamepad gamepadState;
+            gamepadState.buttons = (XInputButton)BitConverter.ToUInt16(bytes, 6);
+            gamepadState.leftTrigger = bytes[8]; 
+            gamepadState.rightTrigger = bytes[9]; 
+            gamepadState.leftStickX = (short)BitConverter.ToInt16(bytes, 10);
+            gamepadState.leftStickY = (short)BitConverter.ToInt16(bytes, 12);
+            gamepadState.rightStickX = (short)BitConverter.ToInt16(bytes, 14);
+            gamepadState.rightStickY = (short)BitConverter.ToInt16(bytes, 16);
```

Comment:
BitConverter methods assume the current system's endianness, so this code won't work in the (theoretical) case that the host system is big-endian. In practice I don't think that would ever happen in the context of Unity lol, but it's generally best not to assume that.

This can be addressed by manually reading the bytes and using bitwise operations to place them in the correct bits:

```suggestion
            gamepadState.buttons = (XInputButton)(bytes[6] | (bytes[7] << 8));
            gamepadState.leftTrigger = bytes[8]; 
            gamepadState.rightTrigger = bytes[9]; 
            gamepadState.leftStickX = (short)(bytes[10] | (bytes[11] << 8));
            gamepadState.leftStickY = (short)(bytes[12] | (bytes[13] << 8));
            gamepadState.rightStickX = (short)(bytes[14] | (bytes[15] << 8));
            gamepadState.rightStickY = (short)(bytes[10] | (bytes[17] << 8));
```